### ISSUE TITLE
consumption-related fields - special cases of bulk products

### ISSUE THREAD

**Issue by serpico** [openfoodfacts/open-prices-frontend#1090]
In relation with #974 and after #1071 it feels like we'll need an evo to display and take into account the amount of products bought and inputted under Category.

In the screenshot I bought around 3.3 kg but there's no field yet to input that value, so for now the chip display is off ( for clarity and completeness, there's another product ( sold per item ) on that proof but it's missing from the list of bulk products available for now ).

If we want user to be able to keep track of all the bulk products they bought over a period of time.

There's also something in connection with discounted price here - I bought it discounted on account of buying by 2 kg ( cheaper than per 1 kg but more expensive than per 10 kg - ideally I'd want to able to store that info and be able to make use of it later ).

![Image](https://github.com/user-attachments/assets/ca21163c-f29c-40d4-b4b6-efc2823faaaf)

**Comment by raphodn** [comment]
Good catch ! Indeed for now we ask for price data, not actual "amount bought" data.
I'll update #974 and see if it can be integrated as is 🤔

**Comment by raphodn** [comment]
So in #974 we added the Product.receipt_quantity field. But in the backend it only allowed "Positive integer" numbers.

I changed the backend here - https://github.com/openfoodfacts/open-prices/pull/666 - to now allow any "Positive decimal" numbers !

![Image](https://github.com/user-attachments/assets/b6f9bf92-6fed-4c64-a0be-128e918873af)

**Comment by serpico** [comment]
So for bulk product the UI will look like Abricots in the screenshot above ? what about just plain weight for products sold by weight ? 2500g or 2.5 kg in the above example.

**Comment by raphodn** [comment]
You're still talking about receipts right ?

Improvements we could do, in line with your comment (correct me if I'm wrong !)
- for the "Quantité achetée" field
  - if its a barcode-product: rename "Quantité achetée" to "Quantité achetée (unité)" and accept only integers
  - if its a category-product with a price per kg: rename "Quantité achetée" to "Quantité achetée (kg)" and accept decimals
  - if its a category-product with a price per unit: rename "Quantité achetée" to "Quantité achetée (unité)" and accept only integers
- for the display in the "X prix déjà téléversés" (and elsewhere in the close future)
  - replace "x2" by "x2 unité"
  - replace "x2.5" by "x2.5 kg"
  - add an extra line indicating the actual amount bought

**Comment by serpico** [comment]
Thanks, that sounds about right ( optional display in grams instead of kg - but I agree it's cosmetic bordering OCD 😄 )

**Comment by raphodn** [comment]
So on some receipts the quantity bought requires a decimal with 3 numbers behind the decimal point (currently limited to 2)

![Image](https://github.com/user-attachments/assets/f834def0-acb9-48ba-80e2-124096988e40)

**Comment by raphodn** [comment]
created an issue in the backend : https://github.com/openfoodfacts/open-prices/issues/794
